updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -21,8 +21,14 @@ def perform_job
         @base_commit_sha = file_fetcher.commit
         raise "base commit SHA not found" unless @base_commit_sha
 
-        version = file_fetcher.package_manager_version
-        api_client.record_package_manager_version(version[:ecosystem], version[:package_managers]) unless version.nil?
+        # We don't set this flag in GHES because there's no point in recording versions since we can't access that data.
+        # TODO: The flag is named `record_ecosystem_versions` because `package_manager_version` is getting renamed to
+        # to that shortly... but splitting into separate PR's for lower risk/easiery testability. Once the follow-on PR
+        # with the rename lands, the name confusion will disappear.
+        if Experiments.enabled?(:record_ecosystem_versions)
```

Comment:
Nothing to do with this change, but I wonder if we should eventually split `Experiments` into `Features` for long-lived switches and `Experiments` for short-term ones 🤔

Reply:
Totally agree. I think the internal repo that controls setting these flags will be the best place to discuss a potential refactoring (the result of which would then flow out here into core).